src/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+
+/**
+ * The XSLTInjection plugin identifies XSLT injection vulnerabilities with all the parameters in GET and POST
+ *
+ * @author CaptainFreak
+ */
+public class XSLTInjection extends AbstractAppParamPlugin {
+
+	// wasc_23 is XML Injection
+	private static Vulnerability vuln = Vulnerabilities.getVulnerability("wasc_23");
+    
+	// plugin dependencies
+    private static final String[] dependency = {};
+    
+	//For logging
+	private static Logger log = Logger.getLogger(XSLTInjection.class);
+    private boolean debugEnabled = log.isDebugEnabled();
+
+    // Payloads
+    
+    private static final String[] errorCausingPayloads={"<",">","\'","\""};
+    
+    private static final String[] vendorReturningPayloads={"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/>",
+    														"system-property(\'xsl:vendor\')/>",
+    														"\"/><xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--",
+    														"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--"
+    														};
+    private static final String[] xsltVendors={"libxslt","Microsoft","Saxonica","Apache","Xalan","SAXON","Transformiix"};
+    
+    //TODO Add time based payloads (SSRF)
+    
+	@Override
+	public int getId() {
+		return 90017;
```

Comment:
You'll need to do a separate PR to reserve the plugin ID.

https://github.com/zaproxy/zaproxy/blob/develop/src/doc/scanners.md

Reply:
I'd suggest 40033 or 20020

Reply:
Is that for a special reason?

Reply:
No they just seemed like the next logical numbers. Honestly the groupings have largely been a mystery to me (I know there was, or is, some passive/active grouping, but outside of that I dunno). 90017 is fine too, I just figured there was no reason to be top-end-heavy..... <shrug>

Reply:
OK. (Yeah, there was some groups/categories from Paros days but they were not followed always.)